core/src/hir/lowering.rs

```diff
@@ -652,7 +653,22 @@ impl<'ast> LoweringContext<'ast> {
                 self.lower_method(method, in_path, attrs, self_id, special_method_presence);
             match (method, &mut methods) {
                 (Ok(method), Ok(methods)) => {
-                    methods.push(method);
+                    if matches!(
+                        method.attrs.special_method,
+                        Some(SpecialMethod::Constructor)
+                    ) {
+                        if !has_unnamed_constructor {
+                            methods.push(method);
+                            has_unnamed_constructor = true;
+                        } else {
+                            self.errors.push(LoweringError::Other(format!(
+                                "At most one unnamed constructor is allowed (extra abi_name: {})",
```

Comment:
nit: add a link to https://github.com/rust-diplomat/diplomat/issues/234 in the error message